crates/modules/smtp-notifier/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+use std::{default::Default, error::Error, fmt, str::FromStr};
+
+use pulsar_core::event::Threat;
+use pulsar_core::pdk::{
+    CleanExit, ConfigError, ModuleConfig, ModuleContext, ModuleError, PulsarModule, ShutdownSignal,
+    Version,
+};
+
+use lettre::{
+    transport::smtp::{
+        authentication::Credentials,
+        client::{Tls, TlsParameters},
+    },
+    AsyncSmtpTransport, AsyncTransport, Message, Tokio1Executor,
+};
+
+const MODULE_NAME: &str = "smtp-notifier";
+
+pub fn module() -> PulsarModule {
+    PulsarModule::new(
+        MODULE_NAME,
+        Version::parse(env!("CARGO_PKG_VERSION")).unwrap(),
+        smtp_notifier_task,
+    )
+}
+
+async fn smtp_notifier_task(
+    ctx: ModuleContext,
+    mut shutdown: ShutdownSignal,
+) -> Result<CleanExit, ModuleError> {
+    let mut receiver = ctx.get_receiver();
+    let mut rx_config = ctx.get_config();
+    let mut config: SmtpNotifierConfig = rx_config.read()?;
+
+    loop {
+        tokio::select! {
+            // Handle configuration changes:
+            _ = rx_config.changed() => {
+                config = rx_config.read()?;
+                continue;
+            }
+            r = shutdown.recv() => return r,
+            msg = receiver.recv() => {
+                let event = msg?;
+
+                // Check if the even is a threat and send a email if it is
+                if let Some(Threat {
+                    source,
+                    description,
+                    extra: _,
+                }) = &event.header().threat
+                {
+                    let payload = event.payload();
+                    let subject = format!("Pulsar Threat Notification: {source}");
+                    let body = format!("{description}\n Source event: {payload}");
+
+                    let message = Message::builder()
+                        .from("Pulsar <pulsar-threat-notification@gmail.com>".parse()?)
+                        .to(config.receiver.parse()?)
+                        .subject(subject)
+                        .body(body)?;
+
+                    let tls_params = TlsParameters::new(config.server.as_str().into())?;
+
+                    AsyncSmtpTransport::<Tokio1Executor>::relay(config.server.as_str())?
+                        .credentials(Credentials::new(config.user.clone(), config.password.clone()))
+                        .port(config.port)
+                        .tls(
+                            match config.encryption {
+                                Encryption::Plain => Tls::None,
+                                Encryption::Tls => Tls::Wrapper(tls_params),
+                                Encryption::StartTls => Tls::Opportunistic(tls_params),
+                            }
+                        )
+                        .build()
+                        .send(message)
+                        .await?;
```

Comment:
```suggestion
    let mut smtp_transport = match config.encryption {
        Encryption::Plain => {
            AsyncSmtpTransport::<Tokio1Executor>::builder_dangerous(config.server.as_str())?
        }
        Encryption::Tls => AsyncSmtpTransport::<Tokio1Executor>::relay(config.server.as_str())?,
        Encryption::StartTls => {
            AsyncSmtpTransport::<Tokio1Executor>::starttls_relay(config.server.as_str())?
        }
    };

   
    smtp_transport.port(port).build().send(message).await?;
```

for `StartTls`, the option should be  `Tls::Required` instead of `Tls::Opportunistic`. however there are built-in helper methods for different behaviours